modules/@apostrophecms/area/ui/apos/components/AposAreaEditor.vue

```diff
@@ -48,6 +48,7 @@
         :field-id="fieldId"
         :disabled="field && field.readOnly"
         :widget-hovered="hoveredWidget"
+        :non-foreign-widget-hovered="hoveredNonForeignWidget"
```

Comment:
If the currently hovered widget is foreign, then the "hovered non foreign widget" will be its closest non-foreign ancestor.